### PR TITLE
Nightlies page use clickhouse, deprecate rockset

### PR DESCRIPTION
**Pull request by atalman**
Rewrite Nightlies page using clickhouse

### CODE REVIEW

File: torchci/pages/nightlies.tsx [hunk 1]
```diff
@@ -13,13 +13,15 @@ function NightlyJobsRedPanel({
...
 }: {
-  params: RocksetParam[];
+  params: { [key: string]: any };
   repo: string;
 }) {
-  let repo_p = params.find(({ name }) => name == "repo");
+  let repo_p = params.find(
+    ({ name }: { [key: string]: any }) => name == "repo"
+  );
   if (repo_p && repo) repo_p.value = repo;
```
Comment (clee2000) [inline torchci/pages/nightlies.tsx:22]:
```suggestion
  if ("repo" in params && repo) params["repo"] = repo;
```

File: torchci/pages/nightlies.tsx [hunk 2]
```diff
@@ -76,16 +75,16 @@ function ValidationRedPanel({
...
   query_type: string;
 }) {
   const url =
-    `/api/query/nightlies/` +
+    `/api/clickhouse/` +
     query_type +
     `_jobs_red?parameters=${encodeURIComponent(
       JSON.stringify([
-        ...params,
+        params,
```
Comment (clee2000) [inline torchci/pages/nightlies.tsx:87]: The ClickHouse params should be a dictionary/object of "fieldname": "value"
ex timezone: Intl.DateTimeFormat().resolvedOptions().timeZone etc

File: torchci/pages/nightlies.tsx [hunk 3]
```diff
@@ -280,31 +278,6 @@ export default function Page() {
...
-            columns={[
-              { field: "COUNT", headerName: "Count", flex: 1 },
-              { field: "name", headerName: "Name", flex: 4 },
-            ]}
-            dataGridProps={{ getRowId: (el: any) => el.name }}
-          />
-        </Grid>
-
-        <Grid item xs={6} height={ROW_HEIGHT}>
-          <NightlyJobsRedPanel params={timeParams} repo={"text"} />
```
Comment (clee2000) [inline torchci/pages/nightlies.tsx:293]: Did you mean to get rid of this?

File: torchci/pages/nightlies.tsx [hunk 4]
```diff
@@ -181,8 +180,9 @@ export default function Page() {
   const [startTime, setStartTime] = useState(dayjs().subtract(1, "week"));
   const [stopTime, setStopTime] = useState(dayjs());
   const [timeRange, setTimeRange] = useState<number>(7);
+  const { useCH: useClickHouse } = useCHContext();
```
Comment (clee2000) [inline torchci/pages/nightlies.tsx:183]:
```suggestion
```
Not used?

File: torchci/pages/nightlies.tsx [hunk 5]
```diff
@@ -1,10 +1,10 @@
 import { Grid, Paper, Skeleton, Stack, Typography } from "@mui/material";
-import TablePanel from "components/metrics/panels/TablePanelRS";
+import TablePanel from "components/metrics/panels/TablePanel";
+import { useCHContext } from "components/UseClickhouseProvider";
```
Comment (clee2000) [inline torchci/pages/nightlies.tsx:3]:
```suggestion
```
not used?

File: torchci/pages/nightlies.tsx [hunk 6]
```diff
@@ -181,8 +180,9 @@ export default function Page() {
   const [startTime, setStartTime] = useState(dayjs().subtract(1, "week"));
   const [stopTime, setStopTime] = useState(dayjs());
   const [timeRange, setTimeRange] = useState<number>(7);
+  const { useCH: useClickHouse } = useCHContext();
 
-  const timeParams: RocksetParam[] = [
+  const timeParams = [
```
Comment (clee2000) [inline torchci/pages/nightlies.tsx:185]: See one of the other comments about param format

File: torchci/pages/nightlies.tsx (hunk 3 above)
Comment (atalman) [reply torchci/pages/nightlies.tsx:293]: Yes text repo is no longer in development

File: torchci/pages/nightlies.tsx (hunk 2 above)
Comment (clee2000) [reply torchci/pages/nightlies.tsx:86]: A dictionary/object, no arrays

So this one would just be JSON.stringify({...params, channel})

File: torchci/pages/nightlies.tsx [hunk 7]
```diff
@@ -218,7 +203,7 @@ export default function Page() {
 
       <Grid container spacing={2}>
         <Grid item xs={6} height={ROW_HEIGHT}>
-          <NightlyJobsRedPanel params={timeParams} repo={"pytorch"} />
+          <NightlyJobsRedPanel params={...timeParams} repo={"pytorch"} />
```
[the remark this answers is not in the document]
Comment (clee2000) [reply torchci/pages/nightlies.tsx:206]: NVM it does not, but the `...` is not useful anyways

File: torchci/pages/nightlies.tsx [hunk 8]
```diff
@@ -182,23 +176,11 @@ export default function Page() {
...
-    },
-    {
-      name: "repo",
-      type: "string",
-      value: "pytorch",
-    },
-  ];
+  const timeParams = {
+    startTime: startTime,
+    stopTime: stopTime,
```
Comment (clee2000) [inline torchci/pages/nightlies.tsx:181]:
```suggestion
    stopTime: dayjs(stopTime).utc().format("YYYY-MM-DDTHH:mm:ss.SSS"),
```

Comment (clee2000) [reply torchci/pages/nightlies.tsx:181]: Might also need to import utc and apply it to dayjs
```
import utc from "dayjs/plugin/utc";
dayjs.extend(utc);
```

File: torchci/pages/nightlies.tsx [hunk 9]
```diff
@@ -71,134 +72,17 @@ function NightlyJobsRedPanel({
   );
 }
 
-function ValidationRedPanel({
```
Comment (atalman) [inline torchci/pages/nightlies.tsx:74]: I believe this is not really required anymore

File: torchci/pages/nightlies.tsx [hunk 10]
```diff
@@ -400,7 +200,6 @@ export default function Page() {
             }
             queryName={"nightly_jobs_red_by_platform"}
             queryParams={timeParams}
```
Comment (clee2000) [inline torchci/pages/nightlies.tsx:202]: The title of this panel and what it is actually querying don't match up: timeParams contains the repo parameter, which is currently set to pytorch, so the query went from looking for pytorch and domains to just pytorch.  I don't know if this is intentional

File: torchci/clickhouse_queries/nightly_jobs_red_past_day/query.sql [hunk 11]
```diff
@@ -1,20 +1,18 @@
...
   AND workflow.name NOT IN ('cron', 'Bandit', 'tests')
   AND push.ref = 'refs/heads/nightly'
-  AND push.repository.owner.name = 'pytorch'
-  AND push.repository.name = :repo
-  AND job.conclusion in ('failure', 'timed_out', 'cancelled') 
-  AND job._event_time >= CURRENT_DATE() - INTERVAL 1 DAY
+  AND push.repository.'owner'.'name' = 'pytorch'
+  AND push.repository.'name' = {repo: String }
+  AND job.conclusion IN ('failure', 'timed_out', 'cancelled')
+  AND job.completed_at >= today() - 1
```
Comment (clee2000) [inline torchci/clickhouse_queries/nightly_jobs_red_past_day/query.sql:16]: Minor note that today() doesn't include hour so you're getting more than 24 hours of data here, don't know if it matters though

**Review by clee2000 — APPROVED** [review]
A few minor comments.  

Some charts look different between rockset + clickhouse but I think its just because of the _event_time -> created_at changes 

I'd also appreciate if you put that you're deleting some unneeded charts in the PR body because I got a bit confused about that

File: torchci/pages/nightlies.tsx (hunk 10 above)
Comment (atalman) [reply torchci/pages/nightlies.tsx:202]: yes just pytorch should be enough here

File: torchci/clickhouse_queries/nightly_jobs_red_past_day/query.sql (hunk 11 above)
Comment (atalman) [reply torchci/clickhouse_queries/nightly_jobs_red_past_day/query.sql:16]: I think its ok we just want to have a list that failed lately

**Pull request merged**